api/middlewares/map-exceptions.ts

```diff
@@ -0,0 +1,107 @@
+import type { ErrorHandler } from "hono";
+import { createMiddleware } from "hono/factory";
+import { HttpStatusCode } from "../helpers/http-code.ts";
+import {
+  BaseException,
+  UniqueConstraintException,
+  ValidationException,
+} from "@sffvektor/lib";
+import type { ContentfulStatusCode } from "hono/utils/http-status";
+import { HTTPException } from "hono/http-exception";
+
+// deno-lint-ignore no-explicit-any
+export type ExceptionConstructor = new (...args: any[]) => Error;
+
+export type ExceptionRecord = Record<string, ContentfulStatusCode>;
+
+export type ApiError = {
+  message: string;
+  code?: string;
+  details?: unknown;
+  stacktrace?: string;
+};
+
+class UnknownInternalError extends BaseException {
+  constructor() {
+    super("Unknown internal error", "UNKNOWN_INTERNAL_ERROR");
+  }
+}
+
+const defaultExceptionMap: ExceptionRecord = {
+  [UniqueConstraintException.name]: HttpStatusCode.Conflict,
+  [ValidationException.name]: HttpStatusCode.BadRequest,
+};
+
+/**
+ * This middleware is meant to be used once globally.
+ */
+export const handleExceptionMiddleware: ErrorHandler = (error, c) => {
+  if (error instanceof HTTPException) {
+    return error.getResponse();
+  }
+  const exceptionsMap = c.get("exceptionsMap");
+  const exceptionHttpStatus = exceptionsMap?.[error?.constructor?.name] ??
+    defaultExceptionMap[error?.constructor?.name] ??
+    HttpStatusCode.InternalServerError;
+  let exception: BaseException;
+  if (error instanceof HTTPException) {
```

Comment:
I don't really understand this 😕 There was already an if in the beginning of the function exactly for the case when `error instanceof HTTPException`, and we return from there, so at this point, it cannot happen, no?